README.md

```diff
@@ -731,6 +731,147 @@ using (DiagnosticPipeline pipeline = DiagnosticPipelineFactory.CreatePipeline(co
+```csharp
+public interface IOutput
+{
+    Task SendEventsAsync(IReadOnlyCollection<EventData> events, long transmissionSequenceNumber, CancellationToken cancellationToken);
+}
+```
+The output receives a batch of events, along with transmission sequence number and a cancellation token. The transmission sequence number can be treated as an identifier for the `SendEventsAsync()` method invocation; it is guaranteed to be unique for each invocation, but there is no guarantee that there will be no "gaps", nor that it will be stricly increasing. The cancellation token should be used to cancel long-running operations if necessary; typically it is passed as a parameter to asynchronous i/o operations.
+
+### Pipeline structure and threading considerations
+Every EventFlow pipeline can be created imperatively, i.e. in the program code. The structure of every pipeline is reflected in the constructor of the `DiagnosticPipeline` class and is as follows:
+
+1. One or more inputs produce events.
+2. A set of common filters (also known as 'global' filters) does initial filtering.
+3. Then the events are sent to one or more outputs (by cloning them as necessary). Each output can have its own set of filters. A combination of an output and a set of filters associated with that output is called a 'sink'.
+
+EventFlow employs the following policies with regards to concurrency:
+
+1. Inputs are free to call OnNext() on associated observers using any thread.
+2. EventFlow will ensure that only one filter at a time is evaluating any given EventData object. That said, the same filter can be invoked concurrently for different events.
+3. Outputs will invoked concurrently for different batches of data. 
+
+### Using custom pipeline intems imperatively
+The simplest way to use custom EventFlow elements is to create a pipeline imperatively, in code. You just need to create a read-only collections of the inputs, global filters and sinks and pass them to `DiagnosticPipeline` constructor. Custom and standard elements can be combined freely; each of the standard pipeline elements has a public constructor and associated public configuration class and can be created imperatively.
+
+### Creating a pipeline with custom elements using configuration
+To create an EventFlow pipeline with custom elements from configuration each custom element needs a factory. The factory is an object implementing `IPipelineItemFactory<TPipelineItem>` and is expected to have a parameter-less constructor.
+
+The factory's `CreateItem(IConfiguration configuration, IHealthReporter healthReporter)` method will receive a configuration fragment that represents the pipeline item being created. The health reporter is available to report issues in case configuration is corrupt or some other problem occurrs during item creation. The health reporter can also be passed to and used by the created pipeline item.
```

Comment:
I believe 'corrupt' is fine in this case (adjective) see http://www.dictionary.com/browse/corrupt?s=t